File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int n=x;
        __int128 rev=0;
        while(x>0){
            int ld=x%10;
            x=x/10;
            rev=(rev*10)+ld;
        }
        if(n==rev){
            return true;
        }else{
            return false;
        }
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int n=x;
        int sum=0;
        if(x<0){
            return false;
        }

        while(x!=0){
            int digit = x%10;
         sum = sum*10 + digit;
            x = x/10;
        }

        if(sum == n){
            return true;
        }
        return false;
    }
}